src/Umbraco.Infrastructure/Security/DeleteExternalLoginsOnMemberDeletingHandler.cs

```diff
@@ -0,0 +1,30 @@
+using Umbraco.Cms.Core.Events;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Notifications;
+using Umbraco.Cms.Core.Services;
+
+namespace Umbraco.Cms.Infrastructure.Security
+{
+    /// <summary>
+    /// Deletes the external logins for the deleted members. This cannot be handled by the database as there is not foreign keys.
+    /// </summary>
+    public class DeleteExternalLoginsOnMemberDeletingHandler : INotificationHandler<MemberDeletingNotification>
+    {
+        private readonly IExternalLoginWithKeyService _externalLoginWithKeyService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DeleteExternalLoginsOnMemberDeletingHandler"/> class.
+        /// </summary>
+        public DeleteExternalLoginsOnMemberDeletingHandler(IExternalLoginWithKeyService externalLoginWithKeyService)
+            => _externalLoginWithKeyService = externalLoginWithKeyService;
+
+        /// <inheritdoc/>
+        public void Handle(MemberDeletingNotification notification)
```

Comment:
Wouldn't it be better to use `MemberDeletedNotification` here instead, to ensure that the memeber has actually been deleted before we delete the external login provider, since `MemberDeletingNotification` is cancelable?